_posts/2020-05-12-semi-automatic-app-versioning-xamarin-forms.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title: Semi-Automatic App Versioning in Xamarin.Forms
+author: spheroid-
+excerpt: Implementing a reasonable versioning scheme in a cross-platform Xamarin.Forms project
+tags:
+- Xamarin
+- iOS
+- Android
+- Git
+---
+
+After you've set up a fresh Xamarin.Forms project (_or a Solution_, as they say), you quickly learn that there's not just one place to adjust the app version, but many, and in some places they even ask for multiple _different_ versions. So how to tame the version number jungle in a reasonable way? I'll show you how I've done it, but first, some basics:
+
+Both Android & iOS apps have two different version numbers we need to care about:
+
+1. The user-visible version string: [CFBundleShortVersionString](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring) on iOS and [versionName](https://developer.android.com/studio/publish/versioning) on Android. On iOS it's _a string of three period-separated integers_ but Android is more liberal about it. To keep things simple, we can go with the iOS format on both platforms. This is what I'll mean with **version number** later on.
+
+2. The actual internal version that the system uses to determine whether _version A_ is greater than _version B_. This is not visible to the user so the format here does not matter much – iOS uses a similarly formatted [CFBundleVersion](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion) as above and on Android it's a single integer [versionCode](https://developer.android.com/studio/publish/versioning). We can, however, use the same format on both platforms as the iOS version can be abbreviated to a single integer. We'll call this one a **build number**.
+
+These are the versions that actually matter in the grand scheme of things, but a Xamarin.Forms project has even more versions (assembly versions, project versions and the solution version to name a few). What should we do with them?
+
+### My Prequisites
```

Comment:
I'm sure it's spelled _prerequisites_

Reply:
It is. I ran the entire post through spell check and corrected a couple of more errors as well.